### PR TITLE
Add master server unit test

### LINKED ISSUE #3079: Add unit test with etcd embed.

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #3079

### CODE REVIEW

File: go/master/service_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		t.Fatal(nil)
+	}
+	cfg := embed.NewConfig()
+	cfg.Dir = etcdDir
+	e, err := embed.StartEtcd(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		e.Close()
```
Comment (helinwang) [inline go/master/service_test.go]: Need to handle error, otherwise linter check would not pass.

File: go/master/service_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		if err := os.RemoveAll(etcdDir); err != nil {
+			log.Fatal(err)
```
Comment (helinwang) [inline go/master/service_test.go]: Could use `t.Fatal(err)` here.

File: go/master/service_test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		}
+	}()
+	select {
+	case <-e.Server.ReadyNotify():
+		log.Printf("Server is ready!")
```
Comment (helinwang) [inline go/master/service_test.go]: Maybe `t.Log` would be better (prints only when test failed or `-v` is specified). Same for other places.

Comment (Yancey1989) [inline go/master/service_test.go]: Done.

File: go/master/service_test.go (hunk 1 above)
Comment (Yancey1989) [inline go/master/service_test.go]: It seems that [Close()](https://godoc.org/github.com/coreos/etcd/embed#Etcd.Close) function return nothing.

File: go/master/service_test.go (hunk 2 above)
Comment (Yancey1989) [inline go/master/service_test.go]: Done.

**Comment by helinwang** [comment]
The develop branch is not building. I think this PR fixes it: https://github.com/PaddlePaddle/Paddle/pull/3106 , please rebase / pull new develop after it gets merged.

**Comment by Yancey1989** [comment]
Thanks @helinwang , fixed the CI failed.

**Pull request merged**